coderd/workspacesearch.go

```diff
@@ -0,0 +1,80 @@
+package coderd
+
+import (
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// WorkspaceSearchQuery takes a query string and breaks it into its queryparams
+// as a set of key=value.
+func WorkspaceSearchQuery(query string) (map[string]string, error) {
+	searchParams := make(map[string]string)
```

Comment:
Since this function is strongly typed, it seems strange to return with an arbitrarily indexed map. Could we return `codersdk.WorkspaceFilter` instead?

Reply:
If I do that, then I need to do the parsing in the function too. 

I know you dislike that you can use query params to assign the `key=value` and use `q=key:value`. In the case it's shared, the code is just easier to reuse if I map the `WorkspaceSearchQuery` to a map[string]string. I can change it to `url.Values`.

---

If we drop supporting query params `?owner=me` then I could return a concrete type more easily.

Reply:
I just dropped query params except `q`, so this is now strongly typed.